### ISSUE TITLE
Docker images docs 

### ISSUE THREAD

**Issue by ksanderer** [buildkite/agent#761]
I have few suggestions about docker images:

1) There should be the docs on the [dockerhub](https://hub.docker.com/r/buildkite/agent/) where explicitly stated base image for tags. Or at least how to determine which base image was used.

2) Agent docker images should have appropriate tag naming for ubuntu and alpine, etc.. Take a look on [python](https://hub.docker.com/_/python/) dockerhub.

    `buildkite/agent:3.1.1-ubuntu`
    `buildkite/agent:3.1.1-alpine`
    `buildkite/agent:3.1.1-alpine3.6`  
etc..

    You can still say that defalut image is alpine. So `buildkite/agent:3.1.1` and `buildkite/agent:3.1.1-alpine` are just symlinks.

IMO it's much more convinient way.

**Why?**
Sometimes it's useful to modify existing image (in my case I need to add `envsubst` to the image). And it was quite confusing for me when [dockerfile](https://hub.docker.com/r/buildkite/agent/~/dockerfile/) dockerhub page use `ubuntu:14.04` and `agent:latest` based on `alpine`.

**Comment by lox** [comment]
Yup, good idea @ksanderer, we're going through a bit of a transition with how we build and manage docker images, so still trying to figure out best practice and how to work around some of the clunky aspets of docker hub.
